roles/network_plugin/calico/tasks/check.yml

```diff
@@ -36,3 +36,27 @@
     msg: "IP in IP and VXLAN mode is mutualy exclusive modes"
   when:
     - "calico_vxlan_mode in ['Always', 'CrossSubnet']"
+
+- name: "Get Calico {{ calico_pool_name }} configuration"
+  command: calicoctl.sh get ipPool {{ calico_pool_name }} -o json
+  failed_when: False
+  changed_when: False
+  register: calico
+  run_once: True
+  delegate_to: "{{ groups['kube-master'][0] }}"
+
+- name: "Set calico_pool_conf"
+  set_fact:
+    calico_pool_conf: '{{ calico.stdout | from_json }}'
+  when: calico.rc == 0 and calico.stdout
+
+- name: "Check if inventory match current cluster configuration"
+  assert:
+    that:
+      - calico_pool_conf.spec.blockSize == (calico_pool_blocksize | default(kube_network_node_prefix))
+      - calico_pool_conf.spec.cidr == (calico_pool_cidr | default(kube_pods_subnet))
+      - calico_pool_conf.spec.ipipMode == calico_ipip_mode
+      - calico_pool_conf.spec.vxlanMode == calico_vxlan_mode
+    msg: "Your inventory doesn't match the current cluster configuration"
+  when:
+    - calico_pool_conf is defined
```

Comment:
This additional check works correctly as https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/882034135
```
TASK [network_plugin/calico : Get Calico default-pool configuration] ***********
task path: /builds/kargo-ci/kubernetes-sigs-kubespray/roles/network_plugin/calico/tasks/check.yml:40
Tuesday 01 December 2020  00:10:01 +0000 (0:00:00.061)       0:12:27.143 ****** 
ok: [instance-1 -> None] => {"changed": false, "cmd": "calicoctl.sh get ipPool default-pool -o json", "failed_when_result": false, "msg": "[Errno 2] No such file or directory: b'calicoctl.sh': b'calicoctl.sh'", "rc": 2}
TASK [network_plugin/calico : Set calico_pool_conf] ****************************
task path: /builds/kargo-ci/kubernetes-sigs-kubespray/roles/network_plugin/calico/tasks/check.yml:48
Tuesday 01 December 2020  00:10:01 +0000 (0:00:00.428)       0:12:27.571 ****** 
TASK [network_plugin/calico : Check if inventory match current cluster configuration] ***
task path: /builds/kargo-ci/kubernetes-sigs-kubespray/roles/network_plugin/calico/tasks/check.yml:53
```